### PR TITLE
Add conformance tests in public ci

### PR DESCRIPTION
**Pull request by Pahandrovich**
This is prototype of conformance tests in regular testing

### CODE REVIEW

**Comment by napetrov** [comment]
discussed offline - let's bring DPC++ support for daal4py build and switch it to conda build. 
here is example that can be reused: 
https://github.com/IntelPython/daal4py/blob/master/azure-pipelines.yml

File: .ci/env/daal4py-dpcpp.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
```
Comment (napetrov) [inline .ci/env/daal4py-dpcpp.sh:1]: I would change name to dpcpp.sh  as from env perspective it's not related to daal4py

File: .ci/env/daal4py-dpcpp.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    intel-oneapi-common-vars      \
+    intel-oneapi-common-licensing \
+    intel-oneapi-tbb-devel        \
+    intel-oneapi-dpcpp-compiler   \
+    intel-oneapi-dev-utilities    \
+    intel-oneapi-libdpstd-devel   \
+    cmake                         \
+    opencl-headers                \
+    libze-loader                  \
+    libze-intel-gpu
```
Comment (napetrov) [inline .ci/env/daal4py-dpcpp.sh:20]: we don't need L0 and opencl headers

File: .ci/pipeline/ci.yml [hunk 3]
```diff
@@ -46,7 +46,45 @@ jobs:
...
+    displayName: 'examples'
+- job: daal4py
+  dependsOn: Linux
+  timeoutInMinutes: 0
+  variables:
+    release.dir: '__release_lnx_gnu'
+  pool:
+    vmImage: 'ubuntu-18.04'
+  strategy:
+    matrix:
```
Comment (napetrov) [inline .ci/pipeline/ci.yml:58]: you can remove strategy section - it's intended for iterating via several configuration sets

File: .ci/pipeline/ci.yml [hunk 4]
```diff
@@ -46,7 +46,45 @@ jobs:
...
+    displayName: clone daal4py
+  - script: |
+      wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O miniconda.sh
+      bash miniconda.sh -b -p $HOME/miniconda
+      export PATH="$HOME/miniconda/bin:$PATH"
+      conda create -q -y -n CB python=3.7 conda-build conda-verify
+    displayName: Create Anaconda environment
+  - script: |
+      export DPCPP_VAR=/opt/intel/inteloneapi/compiler/latest/env/vars.sh
+      export DAALROOT='$(platform.type) build'/daal/latest
```
Comment (napetrov) [inline .ci/pipeline/ci.yml:81]: it should be something like this  DAALROOT="$(Pipeline.Workspace)/$(platform.type)\ build/__release_lnx_gnu/daal/latest"

**Comment by Pahandrovich** [comment]
@Alexsandruss , can you review this PR?

File: .ci/scripts/conformance-scripts/patcher_of_tests.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
+import re
+import sys
+
+alg_name = sys.argv[1]
+
+testToPatch = ['dbscan','elastic_net', 'lin_reg', 'ridge_reg']
+
+case = {
```
Comment (PetrovKP) [inline .ci/scripts/conformance-scripts/patcher_of_tests.py:8]: Why can't I see cases for `svm`, `kmeans` and `pca` ?

File: .ci/scripts/conformance-scripts/run_tests.py [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+        algs = file_algs.read().split("\n")
+    algs.remove("")
+
+    print("Confromance testing start")
+    for alg_name in algs:
+        os.system("bash ./download_tests.sh --alg-name %s --scikit-version 0.23.1 &>> _log_downloads" % alg_name)
+        print(alg_name)
+        os.system("python ./patcher_of_tests.py %s" % alg_name)
+
+        os.system("IDP_SKLEARN_VERBOSE=INFO python -m daal4py -m pytest -s --disable-warnings test_%s.py > _log_%s.txt" % (alg_name,alg_name))
```
Comment (PetrovKP) [inline .ci/scripts/conformance-scripts/run_tests.py:21]: For the future - there is a built-in module [Popen](https://docs.python.org/3/library/subprocess.html) through it, it is easier and more correct to call other processes.

File: .ci/scripts/conformance_test.sh [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+        exit 1
+        ;;
+    esac
+    shift
+    shift
+done
+
+conda create -y -n conformance python=3.7
+source activate conformance
+conda install -y -c conda-forge mpich tbb-devel numpy pytest scikit-learn
```
Comment (PetrovKP) [inline .ci/scripts/conformance_test.sh:21]: Why `conda-forge` not `intel`?

File: .ci/scripts/conformance-scripts/run_tests.py [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+report_filename = "report.html"
+
+if __name__ == "__main__":
+    with open(algs_filename, "r") as file_algs:
+        algs = file_algs.read().split("\n")
+    algs.remove("")
+
+    print("Confromance testing start")
+    for alg_name in algs:
+        os.system("bash ./download_tests.sh --alg-name %s --scikit-version 0.23.1 &>> _log_downloads" % alg_name)
```
Comment (PetrovKP) [inline .ci/scripts/conformance-scripts/run_tests.py:17]: You need version make as parameter, for custom call this sctpts

File: .ci/scripts/conformance-scripts/patcher_of_tests.py (hunk 5 above)
Comment (Pahandrovich) [reply .ci/scripts/conformance-scripts/patcher_of_tests.py:8]: In tests for some algorithms import going from module, but for other import going from specific file. This patch needed for this case.

**Comment by napetrov** [comment]
testing failure can be addressed here - https://github.com/IntelPython/daal4py/pull/226

**Comment by Pahandrovich** [comment]
@PetrovKP, @napetrov, @Alexsandruss, @SmirnovEgorRu, 
can you review this PR?

**Pull request merged**